File: coresuite-dotnet-standard-cs/merger/FormFieldReader.cs
using ceTe.DynamicPDF;
using ceTe.DynamicPDF.Merger;
using ceTe.DynamicPDF.Merger.Forms;
using ceTe.DynamicPDF.PageElements;

namespace coresuite_dotnet_standard_cs.merger
{
    public class FormFieldReader
    {
        public static void Run(string outputPdfPath)
        {
            // PDF to read form fields from
            PdfDocument pdfDocument = new PdfDocument(Util.GetResourcePath("PDFs/fw9AcroForm_18_filled.pdf"));

            // Create Table to display the form fields and values
            Table2 table = new Table2(0, 0, 512, 676, Font.Helvetica, 12);
            table.Border.Width = 1;
            table.Border.Color = RgbColor.Black;
            table.RepeatColumnHeaderCount = 1;
            table.RepeatRowHeaderCount = 1;
            BuildTable(table);
            CreateList(table, pdfDocument.Form.Fields);

            Document document = new Document();
            AddTableToPage(document, table);

            Table2 overflowRowTable = table.GetOverflowRows();
            while (overflowRowTable != null)
            {
                AddTableToPage(document, overflowRowTable);
                overflowRowTable = overflowRowTable.GetOverflowRows();
            }
            document.Draw(outputPdfPath);
        }

        private static void BuildTable(Table2 table)
        {
            CreateColumns(table);
            CreateRowHeadings(table);
        }

        private static void CreateList(Table2 table, PdfFormFieldList fieldList)
        {
            for (int i = 0; i < fieldList.Count; i++)
            {
                Row2 row = table.Rows.Add(20);
                row.Cells.Add(fieldList[i].FullName);
                row.Cells.Add(GetFieldType(fieldList[i]));
                row.Cells.Add(fieldList[i].GetValue());
                if (fieldList[i].HasChildFields == true)
                {
                    CreateList(table, fieldList[i].ChildFields);
                }
            }
        }

        private static string GetFieldType(PdfFormField pdfFormField)
        {
            if (pdfFormField is PdfTextField)
            {
                return "Text Field";
            }
            if (pdfFormField is PdfButtonField)
            {
                return "Button Field";
            }
            if (pdfFormField is PdfChoiceField)
            {
                return "Choice Field";
            }
            if (pdfFormField is PdfSignatureField)
            {
                return "Signature Field";
            }
            return "Container Field";
        }

        private static void AddTableToPage(Document document, Table2 table)
        {
            Page page = new Page(PageSize.Letter);
            if (table != null)
                page.Elements.Add(table);
            document.Pages.Add(page);
        }

        private static void CreateRowHeadings(Table2 table)
        {
            Row2 row = table.Rows.Add(40, Font.TimesBold, 12, RgbColor.Black, RgbColor.LightGrey);
            row.CellDefault.Align = TextAlign.Center;
            row.CellDefault.VAlign = VAlign.Top;
            row.Cells.Add("FormField Name");
            row.Cells.Add("FormField Type");
            row.Cells.Add("FormField Value");
        }

        private static void CreateColumns(Table2 table)
        {
            table.Columns.Add(150);
            table.Columns.Add(150);
            table.Columns.Add(212);
        }
    }
}

File: coresuite-dotnet-framework-console-vb/merger/FormFieldReader.vb
Imports ceTe.DynamicPDF
Imports ceTe.DynamicPDF.Merger
Imports ceTe.DynamicPDF.Merger.Forms
Imports ceTe.DynamicPDF.PageElements

Public Class FormFieldReader

    Public Shared Sub Run(outputPdfPath As String)
        Dim PdfDocument As PdfDocument = New PdfDocument(Util.GetResourcePath("PDFs/fw9AcroForm_18_filled.pdf"))
        'Create Table to display the form fields and values
        Dim MyTable As Table2 = New Table2(0, 0, 512, 676, ceTe.DynamicPDF.Font.Helvetica, 12)
        MyTable.Border.Width = 1.0F
        MyTable.Border.Color = Grayscale.Black
        MyTable.RepeatColumnHeaderCount = 1
        MyTable.RepeatRowHeaderCount = 1

        BuildTable(MyTable)

        CreateList(MyTable, PdfDocument.Form.Fields)

        Dim MyDocument As Document = New Document
        AddTableToPage(MyDocument, MyTable)

        Dim OverflowRowTable As Table2 = MyTable.GetOverflowRows()

        While Not OverflowRowTable Is Nothing
            AddTableToPage(MyDocument, OverflowRowTable)
            OverflowRowTable = OverflowRowTable.GetOverflowRows()
        End While

        MyDocument.Draw(outputPdfPath)
    End Sub

    Private Shared Sub CreateList(ByVal MyTable As Table2, ByVal FieldList As ceTe.DynamicPDF.Merger.Forms.PdfFormFieldList)
        Dim I As Integer
        For I = 0 To FieldList.Count - 1
            ' Dim fieldType As String = fieldList(I).GetType().ToString()
            Dim MyRow As Row2 = MyTable.Rows.Add(20)
            MyRow.Cells.Add(FieldList(I).FullName)
            MyRow.Cells.Add(GetFieldType(FieldList(I)))
            MyRow.Cells.Add(FieldList(I).GetValue())

            If FieldList(I).HasChildFields = True Then
                CreateList(MyTable, FieldList(I).ChildFields)
            End If
        Next
    End Sub

    Private Shared Function GetFieldType(ByVal PdfFormField As PdfFormField) As String
        If TypeOf PdfFormField Is PdfTextField Then
            Return "Text Field"
        End If

        If TypeOf PdfFormField Is PdfButtonField Then
            Return "Button Field"
        End If

        If TypeOf PdfFormField Is PdfChoiceField Then
            Return "Choice Field"
        End If

        If TypeOf PdfFormField Is PdfSignatureField Then
            Return "Signature Field"
        End If

        Return "Container Field"
    End Function

    Private Shared Sub BuildTable(ByVal MyTable As Table2)
        CreateColumns(MyTable)
        CreateRowHeadings(MyTable)
    End Sub

    Private Shared Sub AddTableToPage(ByVal MyDocument As Document, ByVal MyTable As Table2)
        Dim MyPage As Page = New Page(PageSize.Letter)
        If Not MyTable Is Nothing Then
            MyPage.Elements.Add(MyTable)
        End If

        MyDocument.Pages.Add(MyPage)
    End Sub

    Private Shared Sub CreateRowHeadings(ByVal MyTable As Table2)
        Dim MyRow As Row2 = MyTable.Rows.Add(40, Font.TimesBold, 12, Grayscale.Black, Grayscale.LightGrey)
        MyRow.CellDefault.Align = TextAlign.Center
        MyRow.CellDefault.VAlign = VAlign.Top
        MyRow.Cells.Add("FormField Name")
        MyRow.Cells.Add("FormField Type")
        MyRow.Cells.Add("FormField Value")
    End Sub

    Private Shared Sub CreateColumns(ByVal MyTable As Table2)
        MyTable.Columns.Add(150)
        MyTable.Columns.Add(150)
        MyTable.Columns.Add(150)
    End Sub
End Class

